src/wallet/coinselection.cpp

```diff
@@ -171,6 +168,24 @@ bool SelectCoinsBnB(std::vector<OutputGroup>& utxo_pool, const CAmount& target_v
     return true;
 }
 
+bool SelectCoinsSRD(std::vector<OutputGroup>& utxo_pool, const CAmount& target_value, std::set<CInputCoin>& out_set, CAmount& value_ret)
+{
+    out_set.clear();
+    value_ret = 0;
+
+    CAmount selected_eff_value = 0;
+    Shuffle(utxo_pool.begin(), utxo_pool.end(), FastRandomContext());
+    for (const auto& group : utxo_pool) {
```

Comment:
I know you only pass in positive groups, but I wonder if it could be a good sanity check to add an `Assume(group.effective_value > 0)` here?

Reply:
Done